File: src/main/java/com/myapp/domain/Micropost.java
package com.myapp.domain;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Date;

@Entity
@Table(name = "micropost")
@Data
@NoArgsConstructor
public class Micropost {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER)
    private User user;

    @NotNull
    private String content;

    @NotNull
    @Column(name = "created_at")
    private Date createdAt;

    public Micropost(User user, String content) {
        this.user = user;
        this.content = content;
    }

    public Micropost(String content) {
        this.content = content;
    }

    @PrePersist
    protected void onCreate() {
        createdAt = new Date();
    }

}

File: src/main/kotlin/com/myapp/domain/Micropost.kt
package com.myapp.domain

import com.fasterxml.jackson.annotation.JsonProperty
import com.myapp.generated.tables.records.FeedRecord
import com.myapp.generated.tables.records.MicropostRecord
import java.util.*

data class Micropost(

    override val _id: Long? = null,

    val content: String,

    val createdAt: Date = Date(),

    val user: User,

    @get:JsonProperty("isMyPost")
    val isMyPost: Boolean? = null // null means unknown

) : HasIdentity<Long> {

    constructor(record: MicropostRecord, user: User) : this(
        _id = record.id,
        content = record.content,
        createdAt = record.createdAt,
        user = user
    )

    constructor(record: FeedRecord, user: User) : this(
        _id = record.id,
        content = record.content,
        createdAt = record.createdAt,
        user = user
    )

}